selfdrive/car/__init__.py

```diff
@@ -236,3 +241,23 @@ def update(self, current_value, current_counter):
     self.previous_value = current_value
 
     return self.rate
+
+
+@dataclass(order=True)
+class PlatformConfig:
+  platform_str: str
+  car_info: Union[CarInfo, List[CarInfo]]
+  dbc_dict: DbcDict
+
+  def __hash__(self) -> int:
+    return hash(self.platform_str)
+
+
+class Platforms(str, ReprEnum):
+  config: PlatformConfig
+
+  def __new__(cls, platform_config: PlatformConfig):
```

Comment:
this bit of hackery allows `CAR.BODY` to still be a `StrEnum` for backwards compatibility, until we fully move everything to the new `PlatformConfig`. `CAR.BODY.config` will be the `PlatformConfig`

for example, the fingerprints file and everything that references it currently depends on the fact the CAR.BODY is a string

```
FINGERPRINTS = {
  CAR.BODY: [{
    513: 8, 516: 8, 514: 3, 515: 4
  }],
}
```

Reply:
this allows this PR's fingerprint to be small and we can make these changes in small batches